src/main/groovy/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentSelectionWithCurrent.groovy

```diff
@@ -1,13 +1,48 @@
 package com.github.benmanes.gradle.versions.updates.resolutionstrategy
 
-import groovy.transform.TupleConstructor
+import com.thoughtworks.xstream.mapper.Mapper
+import org.gradle.api.artifacts.ComponentMetadata
 import org.gradle.api.artifacts.ComponentSelection
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
 
-@TupleConstructor(includeFields=true)
-class ComponentSelectionWithCurrent {
+import javax.annotation.Nullable
 
-  final String currentVersion
+class ComponentSelectionWithCurrent implements ComponentSelection {
+
+  ComponentSelectionWithCurrent(String currentVersion, ComponentSelection delegate) {
+    this.currentVersion = currentVersion
+    this.delegate = delegate
+  }
 
-  @Delegate
+  final String currentVersion
   private final ComponentSelection delegate
+
+  ModuleComponentIdentifier getCandidate() {
+    return delegate.candidate
+  }
+
+  @Nullable
+  ComponentMetadata getMetadata() {
+    return delegate.metadata
+  }
+
+  def <T> T getDescriptor(Class<T> descriptorClass) {
+    return delegate.getDescriptor(descriptorClass)
+  }
+
+  void reject(String reason) {
+    delegate.reject(reason)
+  }
+
+
+  @Override
+  public String toString() {
```

Comment:
You can probably use interpolation, e.g.
```groovy
"ComponentSelectionWithCurrent{group=${candidate.group}...}"
```

Reply:
done